src/bin/enqueue-job.rs

```diff
@@ -1,29 +1,20 @@
-use cargo_registry::util::{cargo_err, AppError, AppResult};
-use cargo_registry::{db, env, tasks};
-use diesel::PgConnection;
+#![deny(clippy::all)]
 
-fn main() -> AppResult<()> {
+use cargo_registry::{db, env, tasks, util::Error};
+use swirl::Job;
+
+fn main() -> Result<(), Error> {
     let conn = db::connect_now()?;
     let mut args = std::env::args().skip(1);
     match &*args.next().unwrap_or_default() {
-        "update_downloads" => tasks::update_downloads().enqueue(&conn),
+        "update_downloads" => Ok(tasks::update_downloads().enqueue(&conn)?),
         "dump_db" => {
             let database_url = args.next().unwrap_or_else(|| env("DATABASE_URL"));
             let target_name = args
                 .next()
                 .unwrap_or_else(|| String::from("db-dump.tar.gz"));
-            tasks::dump_db(database_url, target_name).enqueue(&conn)
+            Ok(tasks::dump_db(database_url, target_name).enqueue(&conn)?)
         }
-        other => Err(cargo_err(&format!("Unrecognized job type `{}`", other))),
+        other => Err(Error::from(format!("Unrecognized job type `{}`", other))),
```

Comment:
You could use `?` instead of `Error::from` here as well, as you did in the other match arms. (I don't have a strong opinion either way.)

I'm glad you could get rid of the helper trait here. :)

Reply:
It looks like clippy doesn't like `Err(format!("Unrecognized job type `{}`", other))?` (the `clippy::try_err` lint), and the suggestion is to use `.into()`.  I don't have a strong opinion between `Error::from` and `into` here, so I'll leave it as-is.